addons/account/models/account_bank_statement.py

```diff
@@ -295,7 +293,22 @@ def create(self, values):
             next_statements_to_recompute.modified(['previous_statement_id'])
         return res
 
-    def _balance_check(self):
+    @api.depends('line_ids.is_reconciled')
+    def _compute_all_lines_reconciled(self):
+        for statement in self:
+            statement.all_lines_reconciled = all(st_line.is_reconciled for st_line in statement.line_ids)
+
+    @api.onchange('journal_id')
+    def onchange_journal_id(self):
+        self._set_opening_balance(self.journal_id.id)
```

Comment:
Remove this line and you can also remove the following methods which does not seems to be used anymore
def _get_opening_balance(self, journal_id):
def _set_opening_balance(self, journal_id):
def _default_opening_balance(self):